### PR TITLE
[bug] Don't skip column options.

### PR DESCRIPTION
**Pull request by and1truong**

### CODE REVIEW

**Comment by andytruong** [comment]
Very unstable Travis );

**Comment by andytruong** [comment]
@Majkl578 can you review?

File: lib/Doctrine/DBAL/Schema/Column.php [hunk 1]
```diff
@@ -125,14 +125,14 @@ public function setOptions(array $options)
         foreach ($options as $name => $value) {
             $method = "set".$name;
             if ( ! method_exists($this, $method)) {
-                // next major: throw an exception
+                // next major: throw an exception.
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Schema/Column.php]: This change seems unrelated.

File: tests/Doctrine/Tests/DBAL/Schema/ColumnTest.php [hunk 2]
```diff
@@ -65,6 +65,19 @@ public function testSettingUnknownOptionIsStillSupported() : void
         new Column('foo', $this->createMock(Type::class), ['unknown_option' => 'bar']);
     }
 
+    /**
+     * @group legacy
+     * @expectedDeprecation The "unknown_option" column option is not supported, setting it is deprecated and will cause an error in Doctrine 3.0
+     */
+    public function testOptionsShouldNotBeIgnored()
```
Comment (Majkl578) [inline tests/Doctrine/Tests/DBAL/Schema/ColumnTest.php]: Can you please add `void` return type?

File: tests/Doctrine/Tests/DBAL/Schema/ColumnTest.php [hunk 3]
```diff
@@ -65,6 +65,19 @@ public function testSettingUnknownOptionIsStillSupported() : void
...
+    {
+        $col1 = new Column('bar', Type::getType(Type::INTEGER), ['unknown_option' => 'bar', 'notnull' => true]);
+        $this->assertTrue($col1->getNotnull());
```
Comment (Majkl578) [inline tests/Doctrine/Tests/DBAL/Schema/ColumnTest.php]: We use static assertion calls using `self::`.

**Review by Majkl578 — COMMENTED** [review]
LGTM, just few nitpicks. :)

Introduced in #2846.

**Comment by andytruong** [comment]
All fixed per your suggestion. Thanks for fast response @Majkl578.

**Review by greg0ire — APPROVED** [review]
Oh right, looks like I forgot this was in a foreach, sorry for this :(

**Review by morozov — APPROVED** [review]

**Review by Majkl578 — APPROVED** [review]

**Comment by Majkl578** [comment]
@andytruong Thanks!

**Comment by Majkl578** [comment]
Backported into 2.7 @ 2f64799231ecd0e37d9760965c223fcd81b4fecf
